### PR TITLE
feat: load builtin plugins with --clean

### LINKED ISSUE #15605: Make `--clean` act as "factory defaults"
### Neovim version (nvim -v)

NVIM v0.6.0-dev+228-ged0c74778

### Vim (not Nvim) behaves the same?

no (vim 8.2.2671)

### Operating system/version

n/a

### Terminal name/version

n/a

### $TERM environment variable

n/a

### Installation

build from repo

### How to reproduce the issue

1. `nvim --clean`
2. `:scriptnames`
3. `:set rtp?`
4. `:set packpath?`

### Expected behavior

Neovim sets `runtimepath` to `$VIMRUNTIME` **and sources plugins therein**, but omits user config/directories (in effect, behaving like a fresh install whether `.config/nvim` and `.local/share/nvim` didn't exist).

This would make reproducing issues _much_ easier and obviate the need for #15340.

This is pretty much how `vim` behaves, with `:scriptnames` reporting
```
  1: /usr/share/vim/vim82/defaults.vim
  2: /usr/share/vim/vim82/filetype.vim
  3: /usr/share/vim/vim82/ftplugin.vim
  4: /usr/share/vim/vim82/indent.vim
  5: /usr/share/vim/vim82/syntax/syntax.vim
  6: /usr/share/vim/vim82/syntax/synload.vim
  7: /usr/share/vim/vim82/syntax/syncolor.vim
```
and
```
runtimepath=/usr/share/vim/vimfiles,/usr/share/vim/vim82,/usr/share/vim/vimfiles/after
packpath=/usr/share/vim/vimfiles,/usr/share/vim/vim82,/usr/share/vim/vimfiles/after
```

### Actual behavior

Neovim sets `runtimepath` to `$VIMRUNTIME`  but **doesn't source plugins** (and omits user config/directories); `scriptnames` reports nothing.

### PR DESCRIPTION
**Pull request by gpanders**
Closes #15605.

### CODE REVIEW

**Comment by gpanders** [comment]
Not sure if this solution is 100% sufficient, but it seems to work on first glance. Running against CI to find test failures.

**Review by justinmk — APPROVED** [review]
LGTM . Seems like `:help --clean` is vague enough that it doesn't technically need to be updated.

We can iterate on this. Meanwhile, let's do whatever makes `nvim --clean` _more useful to us_ for purposes of minimal bug reports.

**Comment by justinmk** [comment]
Some form of the "expected behavior" from #15605 might be worth adding to `:help --clean` : 

> sets `runtimepath` to `$VIMRUNTIME` **and sources plugins therein**, but omits user config/directories (in effect, behaving like a fresh install as if `.config/nvim` and `.local/share/nvim` didn't exist).

File: runtime/doc/starting.txt [hunk 1]
```diff
@@ -90,10 +90,13 @@ argument.
...
 
 							*--clean*
---clean		Equivalent to "-u NONE -i NONE":
+--clean		Similar to "-u NONE -i NONE":
 		- Skips initializations from files and environment variables.
 		- No 'shada' file is read or written.
 		- Excludes user directories from 'runtimepath'
+		- Unlike "-u NONE -i NONE" plugins are still loaded. This
+		  means "--clean" effectively mimics a fresh install of Nvim
+		  with no user configuration.
```
Comment (gpanders) [inline runtime/doc/starting.txt:99]: cc @justinmk 

Happy to iterate on the wording.

Comment (clason) [reply runtime/doc/starting.txt:99]: "**built-in** plugins are still loaded" or "**default** plugins" (still not ideal, though)

Comment (justinmk) [inline runtime/doc/starting.txt:99]:
```suggestion
		- Loads builtin plugins, unlike "-u NONE -i NONE".
```

File: runtime/doc/starting.txt [hunk 2]
```diff
@@ -90,10 +90,11 @@ argument.
 		See |info-message| about capturing the text.
 
 							*--clean*
---clean		Equivalent to "-u NONE -i NONE":
+--clean		Similar to "-u NONE -i NONE":
```
Comment (justinmk) [inline runtime/doc/starting.txt:93]:
```suggestion
--clean		Mimics a fresh install of Nvim:
```

**Pull request merged**